internal/config/db.go

```diff
@@ -15,6 +15,14 @@ type DatabaseConfig struct {
 }
 
 func (c DatabaseConfig) ToDataSourceName() string {
+	return c.toDataSourceName(true)
+}
+
+func (c DatabaseConfig) ToEmptyDataSourceName() string {
+	return c.toDataSourceName(false)
+}
+
+func (c DatabaseConfig) toDataSourceName(useDBName bool) string {
```

Comment:
I don't think this is the right approach. We already have a check that `c.DatabaseName` is not empty.

If we want to override the config, just create a new struct with the empty field:

```golang
oldConfig := conf.Database

// Copy over existing field values to a new struct
newConfig := config.DatabaseConfig(oldConfig)

// Set database name to empty
newConfig.DatabaseName = ""
```